app/models/user.rb

```diff
@@ -21,34 +21,39 @@ class User < ApplicationRecord
   has_many :badges, through: :badge_achievements
   has_many :collections, dependent: :destroy
   has_many :comments, dependent: :destroy
-  has_many :email_messages, class_name: "Ahoy::Message"
+  has_many :email_messages, class_name: "Ahoy::Message", dependent: :destroy
```

Comment:
I had to specify `dependent` options because otherwise the associated records wouldn't be deleted via `delete_all` (default strategy is `nullify`)
And it doesn't make sense to keep orphaned records in most cases.